src/ssl.c

```diff
@@ -49677,7 +49677,7 @@ static int wolfSSL_TicketKeyCb(WOLFSSL* ssl,
         WOLFSSL_MSG("wolfSSL_HMAC_CTX_Init error");
         return WOLFSSL_TICKET_RET_FATAL;
     }
-    res = ((ticketCompatCb)ssl->ctx->ticketEncCtx)(ssl, keyName,
+    res = ((ticketCompatCtx_t*)ssl->ctx->ticketEncCtx)->cb(ssl, keyName,
```

Comment:
I'm not seeing the point of this. What was the `-pedantic` message that lead to this change?

Reply:
```
src/ssl.c: In function ‘wolfSSL_TicketKeyCb’:
src/ssl.c:49680:12: error: ISO C forbids conversion of object pointer to function pointer type [-Werror=pedantic]
49680 |     res = ((ticketCompatCb)ssl->ctx->ticketEncCtx)(ssl, keyName,
      |            ^
src/ssl.c: In function ‘wolfSSL_CTX_set_tlsext_ticket_key_cb’:
src/ssl.c:49750:25: error: ISO C forbids conversion of function pointer to object pointer type [-Werror=pedantic]
49750 |     ctx->ticketEncCtx = (void*)cb;
      |                         ^
```

Reply:
What if we change `ticketEncCtx` from `void* ticketEncCtx` to
```
union {
  void* ptr;
  ticketCompatCb cb;
} ticketEncCtx;
```
would this satisfy `-pedantic`?

Reply:
That should work... maybe better just to keep them separate. I think if this is being used does it even make sense to allow setting and getting the same data via `wolfSSL_CTX_{s,g}et_TicketEncCtx`?

Reply:
I'm aiming for both. If you think that clarity wise it would be better to add a separate callback field then go for that. One pointer more is not a deal breaker. I wouldn't worry too much about being able to access this through `wolfSSL_CTX_{s,g}et_TicketEncCtx` as its not an OpenSSL API and won't pop up in porting efforts. I don't like the `ticketCompatCtx_t` solution because it drastically reduces clarity (and just feels too much like a hack to satisfy `-pedantic`).

Reply:
Yeah I didn't really like it either. I've opted for keeping them separate. I think it's slightly cleaner. Thanks for the great feedback!